core_impl/src/test/java/com/google/instrumentation/stats/StatsContextFactoryTest.java

```diff
@@ -93,18 +101,34 @@ private static StatsContext testDeserialize(InputStream inputStream) throws IOEx
   }
 
   /*
+   * Construct an InputStream with the given type of tag.
    * The input format is:
    *   <version_id><encoded_tags>, and <encoded_tags> == (<tag_field_id><tag_encoding>)*
    * TODO(songya): after supporting serialize integer and boolean,
    * remove this method and use StatsContext.serialize() instead.
    * Currently StatsContext.serialize() can only serialize strings.
    */
-  private static InputStream constructSingleTypeInputStream(int valueType) throws IOException {
+  private static InputStream constructSingleTypeTagInputStream(int valueType) throws IOException {
     ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-    VarInt.putVarInt(VERSION_ID, byteArrayOutputStream);
-    VarInt.putVarInt(valueType, byteArrayOutputStream);
-    
-    encodeString(KEY1, byteArrayOutputStream);
+    byteArrayOutputStream.write(VERSION_ID);
+    encodeSingleTypeTagToOutputStream(valueType, byteArrayOutputStream);
+    return new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+  }
+
+  // Construct an InputStream with all 4 types of tags.
+  private static InputStream constructMultiTypeTagInputStream() throws IOException {
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    byteArrayOutputStream.write(VERSION_ID);
+    for (int i = 0; i <= VALUE_TYPE_FALSE; i++) {
+      encodeSingleTypeTagToOutputStream(i, byteArrayOutputStream);
+    }
+    return new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+  }
+  
+  private static void encodeSingleTypeTagToOutputStream(
+      int valueType, ByteArrayOutputStream byteArrayOutputStream) throws IOException {
+    byteArrayOutputStream.write(valueType);
+    encodeString(KEY + valueType, byteArrayOutputStream);  // encode <tag_key_len><tag_key>
```

Comment:
Does this line append `valueType` to create a unique tag key?  I think it would help to mention that in a comment.